### PR TITLE
drivers/gpio: added default ISR context struct

### PR DESCRIPTION
**Pull request by haukepetersen**
cleaning up a little more of the GPIO driver code...

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
ACK for the change in general. But I'd like to have 2-3 boards tested before merging. Will do later or ask someone. Maybe @A-Paul?

File: cpu/sam3/periph/gpio.c [hunk 1]
```diff
@@ -43,17 +43,9 @@
...
-typedef struct {
-    gpio_cb_t cb;       /**< callback called from GPIO interrupt */
-    void *arg;          /**< argument passed to the callback */
-} exti_ctx_t;
-
-/**
  * @brief Allocation of memory for 7 independent interrupt slots
  */
-static exti_ctx_t exti_ctx[CTX_NUMOF] = {
+static gpio_isr_ctx_t exti_ctx[CTX_NUMOF] = {
```
Comment (kaspar030) [inline cpu/sam3/periph/gpio.c]: ```= {{0}}; ```
should be enough?
(I know it's unrelated, but now that you're touching it)

**Comment by PeterKietzmann** [comment]
Spot checked on stm43f4 and sam3, was fine. Would you address Kaspars comment?!

**Comment by PeterKietzmann** [comment]
(Will ACK then)

**Comment by haukepetersen** [comment]
Will address Kaspars the comment in a separate PR, so I take your ACK -> and go

**Comment by A-Paul** [comment]
Tested on **cc2538dk** with *tests/periph_gpio*. ISR behaves as intended.

**Pull request merged**